polynote-server/src/main/scala/polynote/server/package.scala

```diff
@@ -174,6 +177,14 @@ package object server {
           }
         }
 
+        override def fetch(path: String): RIO[BaseEnv with GlobalEnv, (String, String)] = {
```

Comment:
Unfortunately I think we should do a check-if-open here. We don't want to force it to open if for some reason there's an independent GET request to download – if that's what causes it to open then it will never close (until someone opens and closes it in the UI)

Reply:
Makes sense. I changed it to be `fetchIfOpen` and to fallback to the filesystem reader response if the notebook is not open.